### PR TITLE
Gherkin and automation of support builds v2 in Dev Console

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-4981
Story: https://issues.redhat.com/browse/ODC-5976

Acceptance criteria:

-  As a developer, I should be able to go to a single nav item (in both Dev & Admin perspective), and see 2 tabs focused on "builds":
    - List view of Build Configs
    - List view of Shipwright Builds
-  As a user, I should be able to see git repository and target image fields when looking at the Shipwright Builds & Build Run details page
-  As a user, I should have access to a dedicated Build Runs page in the Builds section in the Admin perspective
-  As a user, when going to the details page of a Shipwright build, I should be able to access Details, YAML & Build Runs. (Build Runs should have a dedicated tab).
-  As a user, I should be able to access the following actions on a Shipwright Build: Edit YAML, Delete, Edit Annotations, Edit labels
-  As a user, I should be able to access the following actions on a Build Run: Edit YAML, Delete, Edit Annotations, Edit labels

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

**Fix**: https://issues.redhat.com/browse/ODC-6050

**Test Setup**:

Cluster should be running on local

Check the TAGS in frontend/packages/dev-console/integration-tests/cypress.json file be: ""(@pre-condition or @smoke or @regression) and not (@manual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `yarn run dev` and then `npm run test-cypress-dev-console` simultaneously

Execute the build-config/shipwright-build/shipwright-build-detail-page.feature file

**Browser**
Chrome 101

**Test Execution Screenshot:**

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/support/pageObjects/build-po.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  eventTab: '[data-test-id="horizontal-link-public~Events"]',
+  eventStream: '.co-sysevent-stream',
+  breadcrumb: '[aria-label="Breadcrumb"]',
+  failedFilter: '[data-test-row-filter="Failed"]',
+  dev: {
+    buildTab: '[aria-label="Builds"]',
+  },
+  shipwrightBuild: {
+    shipwrightBuildsTab: '[data-test-id="horizontal-link-Shipwright Builds"]',
+    shipwrightBuildRunsTab: '[data-test-id="horizontal-link-BuildRuns"]]',
```
Comment (hemantsaini-7) [inline frontend/packages/dev-console/integration-tests/support/pageObjects/build-po.ts:18]: [data-test-id="horizontal-link-BuildRuns"] single closing bracket

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature [hunk 2]
```diff
@@ -0,0 +1,79 @@
+@odc-4981
+Feature: Shipwright build details page
+              As a user, I want check my Shipwright Build and see all related BuildRuns in a second tab.
+
+        Background:
+            Given user has installed OpenShift Pipelines Operator
+              And user is at developer perspective
+              And user has created or selected namespace "aut-shipwright-build-details"
+              And user has installed Shipwright Operator
+              And user is at Add page
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:10]: Woundn't it make sense to install both operators first and then switch to dev perspective?

```suggestion
            Given user has installed OpenShift Pipelines Operator
              And user has installed Shipwright Operator
              And user is at developer perspective
              And user has created or selected namespace "aut-shipwright-build-details"
              And user is at Add page
```

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+              And user clicks on "Shipwright Builds" tab
+             Then user will see Shipwright Builds
+              And user will see "Succeeded", "Failed" and "Unknown" in Filter list
+
+
+        @smoke
+        Scenario: Shipwright build page in admin perspective: SWB-01-TC02
+            Given user has created shipwright builds
+             When user switches to Administrative perspective
+              And user clicks on Builds tab in Administrative perspective
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:26]:
```suggestion
              And user clicks on Builds navigation in Administrative perspective
```

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+              And user is at developer perspective
+              And user has created or selected namespace "aut-shipwright-build-details"
+              And user has installed Shipwright Operator
+              And user is at Add page
+
+
+        @smoke
+        Scenario: Shipwright build page in dev perspective: SWB-01-TC01
+            Given user has created shipwright builds
+             When user navigates to build tab
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:16]: It's a navigation item, call we them tabs anyway? 

```suggestion
             When user navigates to "Builds"
```

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            Given user has created shipwright builds
+             When user switches to Administrative perspective
+              And user clicks on Builds tab in Administrative perspective
+             Then user will see "Shipwright Builds" tab
+             Then user will see "Shipwright BuildRuns" tab
+
+
+        @regression
+        Scenario: Shipwright build details page: SWB-01-TC03
+            Given user is on build tab in Developer perspective
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:33]: Just one sentence here? Would expect

Given user is in Developer perspective
And user is on the Builds navigation item

?

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+             When user clicks on "Shipwright Builds" tab
+              And user clicks on "buildpack-nodejs-build-heroku" build
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:35]: Precondition is `Given user has created shipwright builds` or ?

**Review by jerolimov — APPROVED** [review]
Added some small ideas / opinions, nothing special or reasons to stop this from getting merged.

Great tests @sanketpathak, big thanks :+1:

Adding approval for the BuildRunStatus change. Please let someone from your team review this for a lgtm.

/approve

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature (hunk 5 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:33]: This is to separate from the build tab in the Administrator perspective and moreover `user is in Developer perspective` is part of the background statement

**Comment by hemantsaini-7** [comment]
Test runs green
/lgtm

image

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/build-config/shipwright-build/shipwright-build-detail-page.feature:10]: I've kept namespace creation before operator installation as post-installation steps for Shipwright Operator include creating Shipwright Build in the working namespace

**Comment by sanketpathak** [comment]
The pr only affects our E2E tests
/label qe-approved
/label docs-approved
/label px-approved

**Pull request merged**